client/src/pages/private/ParticipantTable.js

```diff
@@ -167,8 +169,12 @@ export default () => {
         row.siteName = item?.statusInfos?.[0].data?.siteName;
 
         if (item.statusInfos && item.statusInfos.length > 0) {
-          if (item.statusInfos.find((statusInfo) => statusInfo.status === 'already_hired')) {
-            const previousStatus = item.statusInfos.find((statusInfo) => statusInfo.data?.previous);
+          // Handling already_hired and withdrawn status
+          const previousStatus = item.statusInfos.find((statusInfo) => statusInfo.data?.previous);
```

Comment:
I'm a little hung up on the logic here, but I understand it's an extension of the previous method. This whole method feels like it needs a refactor

Reply:
I am also 100% agree with you. The whole flow is little bit complex, confusing and difficult to understand.